// baseui/file-uploader-beta/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  FileUploaderBeta: true,
  StyledFileRow: true,
  StyledFileRowColumn: true,
  StyledFileRowContent: true,
  StyledFileRowFileName: true,
  StyledFileRowText: true,
  StyledFileRowUploadMessage: true,
  StyledFileRowUploadText: true,
  StyledFileRows: true,
  StyledHint: true,
  StyledImagePreviewThumbnail: true,
  StyledItemPreviewContainer: true,
  StyledLabel: true,
  StyledParentRoot: true,
  StyledTrashCanFilledIconContainer: true
};
Object.defineProperty(exports, "FileUploaderBeta", {
  enumerable: true,
  get: function () {
    return _fileUploaderBeta.default;
  }
});
Object.defineProperty(exports, "StyledFileRow", {
  enumerable: true,
  get: function () {
    return _styledComponents.StyledFileRow;
  }
});
Object.defineProperty(exports, "StyledFileRowColumn", {
  enumerable: true,
  get: function () {
    return _styledComponents.StyledFileRowColumn;
  }
});
Object.defineProperty(exports, "StyledFileRowContent", {
  enumerable: true,
  get: function () {
    return _styledComponents.StyledFileRowContent;
  }
});
Object.defineProperty(exports, "StyledFileRowFileName", {
  enumerable: true,
  get: function () {
    return _styledComponents.StyledFileRowFileName;
  }
});
Object.defineProperty(exports, "StyledFileRowText", {
  enumerable: true,
  get: function () {
    return _styledComponents.StyledFileRowText;
  }
});
Object.defineProperty(exports, "StyledFileRowUploadMessage", {
  enumerable: true,
  get: function () {
    return _styledComponents.StyledFileRowUploadMessage;
  }
});
Object.defineProperty(exports, "StyledFileRowUploadText", {
  enumerable: true,
  get: function () {
    return _styledComponents.StyledFileRowUploadText;
  }
});
Object.defineProperty(exports, "StyledFileRows", {
  enumerable: true,
  get: function () {
    return _styledComponents.StyledFileRows;
  }
});
Object.defineProperty(exports, "StyledHint", {
  enumerable: true,
  get: function () {
    return _styledComponents.StyledHint;
  }
});
Object.defineProperty(exports, "StyledImagePreviewThumbnail", {
  enumerable: true,
  get: function () {
    return _styledComponents.StyledImagePreviewThumbnail;
  }
});
Object.defineProperty(exports, "StyledItemPreviewContainer", {
  enumerable: true,
  get: function () {
    return _styledComponents.StyledItemPreviewContainer;
  }
});
Object.defineProperty(exports, "StyledLabel", {
  enumerable: true,
  get: function () {
    return _styledComponents.StyledLabel;
  }
});
Object.defineProperty(exports, "StyledParentRoot", {
  enumerable: true,
  get: function () {
    return _styledComponents.StyledParentRoot;
  }
});
Object.defineProperty(exports, "StyledTrashCanFilledIconContainer", {
  enumerable: true,
  get: function () {
    return _styledComponents.StyledTrashCanFilledIconContainer;
  }
});
var _fileUploaderBeta = _interopRequireDefault(require("./file-uploader-beta"));
var _styledComponents = require("./styled-components");
var _types = require("./types");
Object.keys(_types).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _types[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _types[key];
    }
  });
});
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
